### PR TITLE
67 torch.compile tests

### PR DESCRIPTION
**Pull request by qiancao**
Added two tests for torch.compile(cox) and torch.compile(weibull). 

Note: the tests were run on CPU-only nodes, GPUs not tested yet.

### CODE REVIEW

**Comment by tcoroller** [comment]
@qiancao ping me when its good for review!

**Comment by tcoroller** [comment]
@qiancao is it worth adding the compiler in the notebooks too? Or this would create issue for windows user?

**Comment by qiancao** [comment]
> @qiancao is it worth adding the compiler in the notebooks too? Or this would create issue for windows user?

@tcoroller We should not alter the existing notebooks to add torch.compile. But in the future, we can consider adding a notebook specifically on how to accelerate the code using torch.compile/TorchScript. I think torch.compile will still be a problem for most windows users at this point:

https://community.intel.com/t5/Blogs/Tech-Innovation/Artificial-Intelligence-AI/Accelerate-PyTorch-Inference-with-torch-compile-on-Windows-CPU/post/1640044
https://pytorch.org/tutorials/prototype/inductor_windows_cpu.html

File: tests/test_torch_compile.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        event = torch.randint(low=0, high=2, size=(N,)).bool()
+        time = torch.randint(low=1, high=100, size=(N,))
+
+        # compiled version of weibull
+        cweibull = torch.compile(weibull)
+
+        loss_weibull = weibull(log_hz, event, time)
+        loss_cweibull = cweibull(log_hz, event, time)
+
+        self.assertTrue(
```
Comment (tcoroller) [inline tests/test_torch_compile.py:65]: @qiancao using Torch.allclose() to prevent converting to numpy. All good else!

**Pull request merged**